app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -29,12 +33,35 @@
     // enum values with DEFAULT actions //
     //////////////////////////////////////
 
-    show_channel_details(R.string.show_channel_details, (fragment, item) ->
-        // For some reason `getParentFragmentManager()` doesn't work, but this does.
-        NavigationHelper.openChannelFragment(
-                fragment.requireActivity().getSupportFragmentManager(),
-                item.getServiceId(), item.getUploaderUrl(), item.getUploaderName())
-    ),
+    show_channel_details(R.string.show_channel_details, (fragment, item) -> {
+        if (isNullOrEmpty(item.getUploaderUrl())) {
+            final int serviceId = item.getServiceId();
+            final String url = item.getUrl();
+            // TODO: Some visual loading indicator
+            ExtractorHelper.getStreamInfo(serviceId, url, false)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(result -> {
+                        NewPipeDatabase.getInstance(fragment.getContext()).streamDAO()
+                                .setUploaderUrl(serviceId, url, result.getUploaderUrl())
+                                .subscribeOn(Schedulers.io()).subscribe();
+                        // For some reason `getParentFragmentManager()` doesn't work, but this does.
+                        NavigationHelper.openChannelFragment(
+                                fragment.requireActivity().getSupportFragmentManager(),
+                                item.getServiceId(), result.getUploaderUrl(),
+                                item.getUploaderName());
+                    }, throwable -> Toast.makeText(
+                            fragment.getContext(),
+                            "Error at Show Channel Details",
```

Comment:
No hardcoded strings, instead they should be declared in the strings.xml and reference here, so that it can be translated from weblate.

Reply:
Fixed in https://github.com/TeamNewPipe/NewPipe/pull/6919/commits/241054fd26a0a04433c3550420daa536e7b0093e